server-spi-private/src/main/java/org/keycloak/authorization/AdminPermissionsSchema.java

```diff
@@ -475,4 +476,40 @@ public Set<String> getScopeAliases(String resourceType, Scope scope) {
 
         return aliases;
     }
+
+    /**
+     * <p>Disables authorization and evaluation of permissions for realm resource types when executing the given {@code runnable}
+     * in the context of the given {@code session}.
+     *
+     * <p>This method should be used whenever a code block should be executed without any evaluation or filtering based on
+     * the permissions set to a realm. For instance, when caching realm resources where access enforcement does not apply.
+     *
+     * @param session the session. If {@code null}, authorization is enabled when executing the code block
+     * @param runnable the runnable to execute
+     */
+    public static void runWithoutAuthorization(KeycloakSession session, Runnable runnable) {
+        if (session == null) {
+            runnable.run();
+            return;
+        }
+
+        try {
+            session.setAttribute(SKIP_EVALUATION, Boolean.TRUE.toString());
+            runnable.run();
+        } finally {
+            session.removeAttribute(SKIP_EVALUATION);
+        }
```

Comment:
This is probably not safe when people make nested calls here as the inner call would remove the session attribute.

Reply:
You could maybe check if it exists, and then call `runnable.run()` directly and return?

Reply:
Yeah, but still, I will need to update those methods on `KeycloakModelUtils` to propagate the attribute, right?

Reply:
@ahus1 Added the logic to skip evaluation if the attribute is already set. Please, let me know if I should also update the propagation when calling `KeycloakModelUtils#runJobInTransaction `.